diff --git a/types/react.d.ts b/types/react.d.ts
index 6c670f2d3bc7487479e910fc205e6b0435059093..9b71a1f1d23aecec2c896cf45167f9fbe375ce89 100644
--- a/types/react.d.ts
+++ b/types/react.d.ts
@@ -1,10 +1,10 @@
-declare module 'virtual:icons/*' {
+declare module 'virtual:icons/*.jsx' {
   import type { SVGProps } from 'react'
   import type React from 'react'
   const component: (props: SVGProps<SVGSVGElement>) => React.ReactElement
   export default component
 }
-declare module '~icons/*' {
+declare module '~icons/*.jsx' {
   import type { SVGProps } from 'react'
   import type React from 'react'
   const component: (props: SVGProps<SVGSVGElement>) => React.ReactElement
